iceoryx_posh/source/mepoo/mem_pool.cpp

```diff
@@ -49,6 +49,7 @@ MemPool::MemPool(const greater_or_equal<uint32_t, CHUNK_MEMORY_ALIGNMENT> chunkS
 {
     if (isMultipleOfAlignment(chunkSize))
     {
+        IOX_EXPECTS(m_chunkSize <= std::numeric_limits<uint64_t>::max() / m_numberOfChunks);
```

Comment:
Nice, thank you.

Very small nit here and in other places. You can use `IOX_EXPECTS_WITH_MSG` to add an explanation of what the failure of this check implies. Like "Requested size of memory pool is greater than the maximum of uint64_t::max()"

Reply:
Oh yes, please. This will make my life easier wenn porting to the new macros which require always an error string

Reply:
Sure
But what do you mean by other places?
I couldn't find any other places around the touched code that would use IOX_EXPECTS and would benefit from the message.
Could you point me to the code you had in mind?

Reply:
Don't worry, if there is a place without a message it will be added when I port the code the the new macros, which I would start once this PR is merged.